### ISSUE TITLE
Mozilla prohibited to update Firefox as a sudo for linux systems

### ISSUE THREAD

**Issue by myBestSoftAndPref** [ghacksuserjs/ghacks-user.js#522]
Mozilla prohibited to update Firefox as a sudo. See [Bug 1323302](https://bugzilla.mozilla.org/show_bug.cgi?id=1323302).

The problem is that the update with ```sudo``` is only way to update Firefox **without downloading whole Firefox distribution** (archive). In case of sudo Firefox update system downloads needed files only.

There is workaround for that, see [this](https://askubuntu.com/a/1039497) answer.
I wrote a small bash script to simplify the Firefox update as the sudo, you can find the script [here](https://github.com/myBestSoftAndPref/soft/tree/master/src/firefox/update-firefox-as-sudo.sh).

To run the script use ```/path-to-script/script.sh /path-to-firefox/firefox``` command.
The ```/path-to-firefox/firefox``` is path to Firefox binary (executable) file.

**Comment by KOLANICH** [comment]
>In case of the sudo a Firefox update system downloads needed files only.

I guess that package managers should support differential updates. And I aggree browsers shouldn't be used as `root`. Downloading 100 MiB of packages is usually not a problem even if you use a slow (for 2018) connection for that.

**Comment by myBestSoftAndPref** [comment]
Do you noticed that often Ubuntu package of Firefox is updated after few days (or event more) after official release? What users should do with vulnerabilities of old Ubuntu Firefox?

What about "package managers" do you tell? In Ubuntu I saw only one Firefox package. Please, can you explain?

**Comment by KOLANICH** [comment]
>Do you noticed that often Ubuntu package of Firefox is updated after few days (or even more) after official release?

Yes, I have noticed that. When I asked on #ubuntu they have claimed that it is OK.

>What about "package managers" do you tell? In Ubuntu I saw only one Firefox package. Please, can you explain?

I mean that if a package manager doesn't support differential updates, it is a bug in a package manager, not firefox. If it supports, but there is no, it is a bug in packaging process, nkt in firefox.

> I think it is too far from normal when you download 100MiB if you can download 6 MiB.

I aggree with you, it is not OK not to have incremental updates and this should be fixed, but anyway, downloading 100 MiB is not a problem nowadays even via slow connections.

**Comment by myBestSoftAndPref** [comment]
I agree that it is can be insecure way compare with package manager or simple replacing of binaries.

My script does not use ```/root``` directory. The script creates temporary root directory and deletes that directory after update. I can not estimate security level of such decision because I don't know linux and Debian-based systems enough. But it is only way to update Firefox through Mozilla's update system that I found.

Anyway, I wanted to show this way to update Firefox and I wanted to see some opinions about that to know how wrong it's. I propose this not for the tutorial but for a consideration of the situation with Firefox update system.

**Comment by Thorin-Oakenpants** [comment]
All over my head (for now), and outside the scope of the repo TBH. Feel free to continue to discuss it, but this is more of a linux forums thing than Firefox's internal workings etc

**Comment by myBestSoftAndPref** [comment]
> All over my head (for now), and outside the scope of the repo TBH. Feel free to continue to discuss it, but this is more of a linux forums thing than Firefox's internal workings etc

Can you explain how the topic can be "more of a linux" if Mozilla don't want to pack Firefox as an AppImage or flatpak or fix its own built-in Firefox update system? Where is "fault" of linux here? :)
It's a problem of Firefox under linux or more exactly under Debian-based systems (e.g. Ubuntu as the most popular linux desktop system).
